envoy/network/socket.h

```diff
@@ -112,6 +112,12 @@ class ConnectionInfoProvider {
    * @return roundTripTime of the connection
    */
   virtual const absl::optional<std::chrono::milliseconds>& roundTripTime() const PURE;
+
```

Comment:
Thanks! The reason is mainly that all the info in ConnectionInfo is related to downstream connection, we can even get the ssl_connection() it that class. downstream_transport_failure can be at that scope I think. I am okay to move, but let's see others' option.

Reply:
Oh, I think I sticked to this `stream_info.downstreamAddressProvider().downstreamTransportFailureReason()` method. if ConnectionInfoProvider/Setter can be shared both upstream and downstream, it makes sense to move out from it. Thanks!

Reply:
I know there are some downstream related code there. However, by nature, it is part of connection and it can be used by both upstream and downstream. right? See enableSettingInterfaceName() in that class which is used by upstream. 

Off top of my head, as I mentioned above, you can have a new class DownstreamInfo and create it in the `ServerConnectionImpl::ServerConnectionImpl` to strictly differentiate them, but I think it is overkill just for adding failure reason.

Reply:
Agreed, Thanks!